src/background/datastore.js

```diff
@@ -91,6 +91,16 @@ export function convertItem2Info(item) {
   return info;
 }
 
+function extractOrigin(str) {
+  try {
+    const url = new URL(str);
+    return url.origin;
+  } catch (ex) {
+    // ignore failure  -- pretend it'll still save
```

Comment:
Hm. If we ignore failure, it seems like the `(new URL(this.state.origin).hostname)` down below might also throw. Am I missing something?

Reply:
Oh, that's a good catch.  The only URLs I have seen throw are `chrome://` URLs, which _shouldn't_ make their way that far into the app, but they could.  I'll move this logic to a more generic helper.